src/main.rs

```diff
@@ -322,26 +325,28 @@ async fn start(
 	)
 	.await;
 
-	let span = info_span!(parent: None, "xdg-desktop-portal-cosmic");
-	let mut sockets = Vec::with_capacity(1);
-	let extra_env = Vec::with_capacity(1);
-	let portal_extras =
-		if let Ok((mut env, fd)) = create_privileged_socket(&mut sockets, &extra_env) {
-			let mut env = env.remove(0);
-			env.0 = "PORTAL_WAYLAND_SOCKET".to_string();
-			vec![(fd, env, sockets.remove(0))]
-		} else {
-			Vec::new()
-		};
-	start_component(
-		XDP_COSMIC.unwrap_or("/usr/libexec/xdg-desktop-portal-cosmic"),
-		span,
-		&process_manager,
-		&env_vars,
-		&socket_tx,
-		portal_extras,
-	)
-	.await;
+	if env::var("XDG_CURRENT_DESKOP").ok().is_some_and(|desktop| desktop == "COSMIC") {
```

Comment:
It has the same effect, but I think ` env::var("XDG_CURRENT_DESKOP").as_deref() == Ok("COSMIC") ` is a bit more concise and readable.